docs/zenofmetpy.md

```diff
@@ -0,0 +1,29 @@
+# Zen and the Art of MetPy Maintenance
+
+One of the core principles of MetPy is to bring the best practices from software engineering
+into the realm of scientific and research software. In this guide we attempt to lay out those
+principles in the clearest way possible.
+
+* Make no unnecessary assumptions - use physics based formula whenever possible over empirical
+  relations.
+* Clearly state all assumptions.
+* Never depend on inputs with certain units.
+* Function and class names should be spelled without abbreviation.
+  (i.e. `equivalent_potential_temperature`, not `theta_e`)
+* A single letter variable name is not acceptable for anything other than a counter. Variables
+  should be spelled without abbrevation. (i.e. `temperature`, not `T` or `temp`)
+* Vectorize calculations where possible - nested loops are the last resort.
+* Generalize as much as possible, but no more.
+* Users come first, API design, discoverability, and configurability are paramount.
+* All contributions should cite the relevant literature -
+  no constants or calculations that simply "appear".
+* "That's how we've always done it" is not a citation.
+* GEMPAK/NSHARP/NCL is not a citation.
+* Slide decks are not a citation.
+* Compare against legacy calculations when possible.
```

Comment:
What does this include? A simple message of "I compared against X", or perhaps a test that reads in output from another program and compares it to the calculation?